### PR TITLE
Add fix for maxEnforced -> maxEnforcement

### PR DESCRIPTION
**Pull request by Piinks**
This adds a fix for the deprecated `maxEnforced` on `CupertinoTextField`, `TextField`, and `TextFormField`
Deprecated in https://github.com/flutter/flutter/pull/72043

Part of #74908 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test_fixes/cupertino.dart [hunk 1]
```diff
@@ -115,4 +115,12 @@ void main() {
...
   CupertinoTextThemeData.resolveFrom(context, nullOk: true);
   CupertinoTextThemeData.resolveFrom(context, nullOk: false);
+
+  // Changes made in https://github.com/flutter/flutter/pull/72043
+  CupertinoTextField(maxLengthEnforced: true);
+  CupertinoTextField(maxLengthEnforced: false);
+  CupertinoTextField.borderless(maxLengthEnforced: true);
+  CupertinoTextField.borderless(maxLengthEnforced: false);
+  final CupertinoTextField textField;
+  textField.maxLengthEnforced;
```
Comment (goderbauer) [inline packages/flutter/test_fixes/cupertino.dart:125]: Isn't this going from a bool to being an enum? That'll break peoples code if we simply replace "maxLengthEnforced" with "maxLengthEnforcement", no?

Comment (Piinks) [reply packages/flutter/test_fixes/cupertino.dart:125]: I think so, but isn't this the same as `autovalidate` -> `autovalidateMode`? bool to enum? https://github.com/flutter/flutter/pull/73996

Comment (goderbauer) [reply packages/flutter/test_fixes/cupertino.dart:125]: Oh, I didn't notice that there. It does seem a little odd that we leave your code broken after the migration.

Comment (goderbauer) [reply packages/flutter/test_fixes/cupertino.dart:125]: Discussed this offline and decided that including these partial fixes is a good way to tell people that their code needs updating for now. Will revisit if user feedback is different.

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**